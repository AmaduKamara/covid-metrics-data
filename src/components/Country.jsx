import { PropTypes } from 'prop-types';
import { BsArrowRightCircle } from 'react-icons/bs';

const Country = ({ country, TotalConfirmed }) => (
  <div className="border border-pink-600 p-5">
    <div className="flex justify-end items-end">
      <BsArrowRightCircle />
    </div>
    <div className="flex flex-col justify-end items-end mt-5">
      <h1 className="text-xs sm:text-lg md:text-xl lg:text-2xl  uppercase text-white">{country}</h1>
      <p className="text-white text-xs sm:text-sm md:text-lg mt-1">{TotalConfirmed}</p>
    </div>
  </div>
);

Country.propTypes = {
  country: PropTypes.string.isRequired,
  TotalConfirmed: PropTypes.string.isRequired,
};

export default Country;
