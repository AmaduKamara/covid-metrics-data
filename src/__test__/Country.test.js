import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Country from '../components/Country';

describe('Country tests', () => {
  test('should render', () => {
    const countryData = { Country: 'Sierra Leone', TotalConfirmed: 343212 };
    const country = render(
      <BrowserRouter>
        <Country country={countryData} />
      </BrowserRouter>,
    );

    expect(country).toMatchSnapshot();
  });
});
