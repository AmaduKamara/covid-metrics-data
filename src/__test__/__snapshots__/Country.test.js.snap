// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country tests should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="border border-pink-400 py-5 country px-8"
      >
        <div
          class="flex justify-end items-end"
        >
          <svg
            class="text-white text-2xl cursor-pointer"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="flex flex-col justify-end items-end mt-8 md:mt-16"
        >
          <button
            class="text-lg sm:text-xl md:text-xl lg:text-2xl font-semibold uppercase text-white focus:outline-none"
            type="button"
          >
            Sierra Leone
          </button>
          <p
            class="text-white text-xs sm:text-sm md:text-lg mt-1"
          >
            343212
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="border border-pink-400 py-5 country px-8"
    >
      <div
        class="flex justify-end items-end"
      >
        <svg
          class="text-white text-2xl cursor-pointer"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="flex flex-col justify-end items-end mt-8 md:mt-16"
      >
        <button
          class="text-lg sm:text-xl md:text-xl lg:text-2xl font-semibold uppercase text-white focus:outline-none"
          type="button"
        >
          Sierra Leone
        </button>
        <p
          class="text-white text-xs sm:text-sm md:text-lg mt-1"
        >
          343212
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
